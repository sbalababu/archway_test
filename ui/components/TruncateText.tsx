import * as React from 'react';

interface Props {
  text: string;
  maxLine: number;
  lineHeight: number;
  style?: React.CSSProperties;
}

// tslint:disable-next-line: no-shadowed-variable
const TruncateText = ({ text, maxLine, lineHeight, style }: Props) => (
  <div
    style={{
      overflow: 'hidden',
      textOverflow: 'ellipsis',
      display: '-webkit-box',
      lineHeight: `${lineHeight}px`,
      minHeight: `${lineHeight * maxLine}px`,
      WebkitLineClamp: maxLine,
      WebkitBoxOrient: 'vertical',
      wordBreak: 'break-all',
      ...style,
    }}
  >
    {text}
  </div>
);

export default TruncateText;
