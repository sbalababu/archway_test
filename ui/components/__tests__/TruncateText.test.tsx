import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import TruncateText from '../TruncateText';

describe('TruncateText', () => {
  it('it renders correctly', () => {
    const props = {
      text: 'truncate',
      maxLine: 2,
      lineHeight: 100,
    };
    const wrapper = shallow(<TruncateText {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
