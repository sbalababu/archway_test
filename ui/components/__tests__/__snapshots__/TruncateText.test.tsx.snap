// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TruncateText it renders correctly 1`] = `
<div
  style={
    Object {
      "WebkitBoxOrient": "vertical",
      "WebkitLineClamp": 2,
      "display": "-webkit-box",
      "lineHeight": "100px",
      "minHeight": "200px",
      "overflow": "hidden",
      "textOverflow": "ellipsis",
      "wordBreak": "break-all",
    }
  }
>
  truncate
</div>
`;
